compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -1023,3 +1023,20 @@ pub fn dump_mir_def_ids(tcx: TyCtxt<'_>, single: Option<DefId>) -> Vec<DefId> {
         tcx.mir_keys(()).iter().map(|def_id| def_id.to_def_id()).collect()
     }
 }
+
+/// Calc converted u64 decimal into hex and return it's length in chars
+///
+/// ```ignore (cannot-test-private-function)
+/// assert_eq!(1, hex_number_length(0));
+/// assert_eq!(1, hex_number_length(1));
+/// assert_eq!(2, hex_number_length(16));
+/// ```
+fn hex_number_length(x: u64) -> usize {
+    let mut length = if x == 0 { 1 } else { 0 };
```

Comment:
if `x ==0`, why not just early return `1`?

Reply:
When i wrote this i remember that this will give the same asm, but i'll recheck.

Reply:
Returning early is more clear to me